### PR TITLE
docs: clarify padding guidance in developer docs

### PR DESCRIPTION
**Pull request by shadow0500**
Update after cleaning some stuff

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/develop/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Fixes: N/A

## Does this PR introduce a breaking change?
```
[ ] Yes
[ ] No
```

## Other information

### CODE REVIEW

**Comment by damienwebdev** [comment]
@xelaint can you take a look at this and verify that it looks correct to you?

**Comment by xelaint** [comment]
@damienwebdev I'm not sure if this doc is necessary to begin with.

**Comment by damienwebdev** [comment]
I like it, but perhaps its part of a much larger doc describing the "reasoning" behind how we decide on certain styles.

However, are you saying that you think it's no longer relevant?

**Comment by shadow0500** [comment]
I am sorry if I did something wrong but I will try my best from the next time and thank you for your feedback.

**Comment by damienwebdev** [comment]
@shadow0500 no worries! I think the doc here looks good, but I'll let @xelaint decide on what changes should be made.

**Comment by shadow0500** [comment]
oh Thankyou , okay I will be waiting for that and look forward to collaborate with you .

File: docs/styles.md [hunk 1]
```diff
@@ -1,3 +1,58 @@
 
 ## Child Component outer Padding/Margin
-Child components should not define their own outer padding or margin. This is to ensure the component can be reused in other parts of an app. The parent component should define the padding and margin between its child components.
\ No newline at end of file
+It's important for child components to not define their own outer padding or margin. This is to make sure that the component can be reused in different parts of an app. Instead, the padding and margin between child components should be defined by the parent component.
```
Comment (xelaint) [inline docs/styles.md:3]:
```suggestion
It's important for individual components to not define their own outer spacing. Components that define their own outer spacing are limited to specific use cases, making it difficult for them to be reusable without overriding their styles with additional CSS. Instead, spacing between components should be handled by the wrapping parent element via the `margin`, `padding`, `gap` or `grid-gap` properties.
```

File: docs/styles.md [hunk 2]
```diff
@@ -1,3 +1,58 @@
 
 ## Child Component outer Padding/Margin
-Child components should not define their own outer padding or margin. This is to ensure the component can be reused in other parts of an app. The parent component should define the padding and margin between its child components.
\ No newline at end of file
+It's important for child components to not define their own outer padding or margin. This is to make sure that the component can be reused in different parts of an app. Instead, the padding and margin between child components should be defined by the parent component. 
+
+Keeping child components modular ensures that they can be used in different contexts. By having the parent component define the spacing between its child components, you can ensure that the child components will maintain their intended layout regardless of where they are used. This can also help to reduce the amount of CSS code needed to style the child components, making your code more efficient and easier to maintain.
```
Comment (xelaint) [inline docs/styles.md:5]:
```suggestion
Keeping components modular ensures that they are flexible and extensible. By allowing the parent element to control the spacing for inner elements, individual components will maintain their intended layout regardless of where they are used. This allows for the codebase to be maintained and updated more easily.
```

File: docs/styles.md [hunk 3]
```diff
@@ -1,3 +1,58 @@
...
+
+Here's an example of how you might structure the HTML and CSS for a parent component and its child components:
```
Comment (xelaint) [inline docs/styles.md:7]:
```suggestion
## Example
```

File: docs/styles.md [hunk 4]
```diff
@@ -1,3 +1,58 @@
...
+
+``` html 
+<div class="parent-component">
```
Comment (xelaint) [inline docs/styles.md:10]:
```
<div class="wrapper">
  <div class="component">
    <h3>Title</h3>
    <p>Description</p>
  </div>
  <div class="component">
    <h3>Title</h3>
    <p>Description</p>
  </div>
  <div class="component">
    <h3>Title</h3>
    <p>Description</p>
  </div>
</div>

```

File: docs/styles.md [hunk 5]
```diff
@@ -1,3 +1,58 @@
...
+  <div class="child-component">
+    <h3>Child Component 3</h3>
+    <p>This is some text for Child Component 3.</p>
+  </div>
+</div>
+
+```
+
+``` CSS
+.parent-component {
```
Comment (xelaint) [inline docs/styles.md:28]:
```suggestion
.wrapper {
```

File: docs/styles.md [hunk 6]
```diff
@@ -1,3 +1,58 @@
...
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  justify-content: center;
+  padding: 20px;
```
Comment (xelaint) [inline docs/styles.md:33]: remove padding and margin, add

`gap: 16px;`

File: docs/styles.md [hunk 7]
```diff
@@ -1,3 +1,58 @@
...
+  background-color: #f5f5f5;
+}
+.child-component {
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  justify-content: center;
+  background-color: #fff;
+  padding: 20px;
+  margin-bottom: 10px;
```
Comment (xelaint) [inline docs/styles.md:44]: remove `margin-bottom`

this style is doing what the doc is saying what not to do.

File: docs/styles.md [hunk 8]
```diff
@@ -1,3 +1,58 @@
...
+.parent-component {
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  justify-content: center;
+  padding: 20px;
+  margin: 10px;
+  background-color: #f5f5f5;
+}
+.child-component {
```
Comment (xelaint) [inline docs/styles.md:37]:
```suggestion
.component {
```

File: docs/styles.md [hunk 9]
```diff
@@ -1,3 +1,58 @@
...
+  margin-bottom: 10px;
+}
+.child-component p {
+  font-size: 16px;
+  line-height: 1.5;
+}
+
+```
+
+In this example, the parent component defines the padding and margin for all of its child components. Each child component only defines its own internal padding and margin. This allows the child components to be reused in other parts of the app without having to worry about their outer spacing.
```
Comment (xelaint) [inline docs/styles.md:58]:
```suggestion
In this example, the wrapper element sets the gaps between the rows and columns of the components in a flex layout. The inner component defines its own internal padding.
```

**Review by xelaint — CHANGES_REQUESTED** [review]

**Comment by damienwebdev** [comment]
@shadow0500 please submit another PR with the requested changes.

**Pull request closed without merging**